projects/batfish/src/main/java/org/batfish/bdp/BdpDataPlanePlugin.java

```diff
@@ -295,6 +438,313 @@ public Answer computeDataPlane(boolean differentialContext) {
     return answer;
   }
 
+  private void computeDependentRoutesIteration(
+      Map<String, Node> nodes,
+      Topology topology,
+      BdpDataPlane dp,
+      int dependentRoutesIterations,
+      SortedSet<Prefix> oscillatingPrefixes) {
+
+    /*
+     * Oscillation recovery data structures
+     */
+    Map<BgpNeighbor, Map<BgpNeighbor, List<BgpAdvertisement>>> deferredBgpAdvertisements =
+        new IdentityHashMap<>();
+    Map<BgpNeighbor, Map<BgpNeighbor, List<BgpRoute>>> deferredIncomingRoutes =
+        new IdentityHashMap<>();
+    Map<BgpNeighbor, Map<BgpNeighbor, List<BgpMultipathRib>>> deferredIncomingRouteRibs =
+        new IdentityHashMap<>();
+    Map<Pair<String, String>, Map<Pair<String, String>, Set<Prefix>>> markedPrefixes =
+        new HashMap<>();
+
+    // (Re)initialization of dependent route calculation
+    AtomicInteger reinitializeDependentCompleted =
+        _batfish.newBatch(
+            "Iteration " + dependentRoutesIterations + ": Reinitialize dependent routes",
+            nodes.size());
+    nodes
+        .values()
+        .parallelStream()
+        .forEach(
+            n -> {
+              for (VirtualRouter vr : n._virtualRouters.values()) {
+
+                /*
+                 * RIBs that are read from
+                 */
+                vr._prevMainRib = vr._mainRib;
+                vr._mainRib = new Rib(vr);
+
+                vr._prevOspfExternalType1Rib = vr._ospfExternalType1Rib;
+                vr._ospfExternalType1Rib = new OspfExternalType1Rib(vr);
+
+                vr._prevOspfExternalType2Rib = vr._ospfExternalType2Rib;
+                vr._ospfExternalType2Rib = new OspfExternalType2Rib(vr);
+
+                vr._prevBgpRib = vr._bgpMultipathRib;
+                vr._bgpMultipathRib = new BgpMultipathRib(vr);
+
+                vr._prevBgpBestPathRib = vr._bgpBestPathRib;
+                vr._prevBgpBestPathRib._prev = null;
```

Comment:
add comment: BestPathRibs save a pointer to the previous RIB in order to implement oldest-route-wins. Once a RIB moves from the "current" RIB to the "previous" RIB, null out its "previous" RIB to not keep full history of RIBs and enable GC.

Consider whether the `null`ing logic has a better place (e.g., in the `importRib` function, .... or somewhere else.)